internal/ingress/controller/kong.go

```diff
@@ -72,8 +74,28 @@ func (n *KongController) OnUpdate(ctx context.Context, state *kongstate.KongStat
 	} else {
 		err = n.onUpdateDBMode(targetContent)
 	}
+	var target []byte
+	if n.cfg.DumpConfig == util.ConfigDumpModeSensitive {
+		target, _ = json.Marshal(targetContent)
+	} else if n.cfg.DumpConfig == util.ConfigDumpModeEnabled {
+		sanitizedState := state.SanitizedCopy()
+		target, _ = json.Marshal(n.toDeckContent(ctx, sanitizedState))
```

Comment:
These `errors`s should not be ignored because `target` is invalid if `err != nil`.

Instead of ignoring error (and `if len(target) > 0` below), let's have `err != nil` here suppress writing.

I think that this can be cleanly achieved by wrapping the whole new dumping logic being introduced here in a function like `func dumpConfig(configDumpMode, targetContent, dumpDir) error`